### ISSUE TITLE
Yes, but what are the acceptable components of a .DEF file?

### ISSUE THREAD

**Issue by anklassen** [MicrosoftDocs/cpp-docs#3834]
I was looking for a grammar of the .DEF file. At the least, something like the railroad diagrams that SQL documentation pages give of each of the commands one can use there, or like the Language Reference (for language lawyers) pages given on python documentation. I expect the whole grammar to be more simple than either but I just wanted to see it laid out like that. I was disappointed.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 597b87dd-e653-55ff-176b-7e5246afa31a
* Version Independent ID: 5af3db37-b761-07cc-bca9-bcccecaee9f3
* Content: [Module-Definition (.Def) Files](https://docs.microsoft.com/en-us/cpp/build/reference/module-definition-dot-def-files?view=msvc-170)
* Content Source: [docs/build/reference/module-definition-dot-def-files.md](https://github.com/Microsoft/cpp-docs/blob/main/docs/build/reference/module-definition-dot-def-files.md)
* Product: **visual-cpp**
* Technology: **cpp-tools**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@anklassen 
That's an interesting idea, but for a couple of reasons, I'm not planning to address it. The first reason is, *`.DEF`* files are considered deprecated in favor of source annotations and linker options. The second reason is, the file format dates back to the late 1980s. The text of the documentation is probably of a similar age, migrated who knows how many times from paper to help files to web articles. It hasn't changed much in the interim, and there hasn't been much call for such info. It doesn't seem to have much value except for, say, people trying to recreate the linker behavior with a parser of their own. You can probably count such people on your thumbs, possibly with a thumb or two left over. With such a limited audience, in a technology that isn't recommended, it's impossible to justify the time.

**Comment by anklassen** [comment]
> @anklassen That's an interesting idea, but ... not planning to address it. ... deprecated ... late 1980s ... [can't] justify the time.

Fair enough -- and actually I solved my "problem" elsewhere, will show up in ANSAKSoftware space as sqlite-packager or something like that; I will enlarge on it volubly if you dare to betray even a particle of interest (I'm probably just as easy to find on LinkedIn as you were, for instance) -- and .DEF files are still important when all you have besides it is a DLL and a header. I was looking for confirmation of a point that I was able to confirm experimentally, so all is well... Isn't it funny how this stuff just lasts and lasts and lasts though. o7, moving on.
